docker/provider.go

```diff
@@ -301,19 +302,24 @@ func convertDockerConfigToAuthConfigs(confs map[string]dockerConfig) (*AuthConfi
 }
 
 // getCredentialsFromOSKeychain get config from system specific keychains
-func getCredentialsFromOSKeychain(registryAddress string) (types.AuthConfig, error) {
+func getCredentialsFromOSKeychain(registryAddress string, credsStore string) (types.AuthConfig, error) {
 	authConfig := types.AuthConfig{}
-	log.Printf("[DEBUG] Getting auth for registry '%s' on OS: '%s'", registryAddress, runtime.GOOS)
-	if runtime.GOOS == "darwin" {
-		p := osx.NewShellProgramFunc("docker-credential-osxkeychain")
-		credentials, err := osx.Get(p, registryAddress)
-		if err != nil {
-			return authConfig, err
-		}
-		authConfig.Username = credentials.Username
-		authConfig.Password = credentials.Secret
-		authConfig.ServerAddress = registryAddress
-		authConfig.Auth = base64.StdEncoding.EncodeToString([]byte(credentials.Username + ":" + credentials.Secret))
+	log.Printf("[DEBUG] Getting auth for registry '%s' from credential store: '%s'", registryAddress, credsStore)
+	if credsStore == "" {
+		return authConfig, errors.New("No credential store configured")
+	}
+	executable := "docker-credential-" + credsStore
```

Comment:
yes because they all follow the same pattern according to https://github.com/docker/docker-credential-helpers#available-programs